### PR TITLE
Fix for issue #2680, root cause map.options is no longer available

### LINKED ISSUE #2680: Fix exception thrown by "drag rotate" handler
per https://github.com/mapbox/mapbox-gl-js/commit/4dd6786282e7d3e9e7b2a7fd866c3637d9e88737#commitcomment-17747349

### PR DESCRIPTION
**Pull request by blanchg**
Now pass all the mapOptions to the constructor of each handler to fix #2680

### CODE REVIEW

File: js/ui/handler/drag_rotate.js [hunk 1]
```diff
@@ -17,9 +17,10 @@ var inertiaLinearity = 0.25,
  * dragging the cursor while holding the right mouse button or the `ctrl` key.
  * @class DragRotateHandler
  */
-function DragRotateHandler(map) {
+function DragRotateHandler(map, options) {
     this._map = map;
     this._el = map.getCanvasContainer();
+    this._options = options;
```
Comment (mourner) [inline js/ui/handler/drag_rotate.js]: Better to explicitly save `this._bearingSnap` here instead of the options object.

**Comment by mourner** [comment]
Nice, looks good to me other than the minor comment above.

File: js/ui/bind_handlers.js [hunk 2]
```diff
@@ -13,15 +13,15 @@ var handlers = {
     touchZoomRotate: require('./handler/touch_zoom_rotate')
 };
 
-module.exports = function bindHandlers(map, options) {
+module.exports = function bindHandlers(map, bindOptions, mapOptions) {
```
Comment (lucaswoj) [inline js/ui/bind_handlers.js]: Why the second `mapOptions` object? Could the `bearingSnap` property go on the original `options`/`bindOptions` object?

**Comment by blanchg** [comment]
It could, but would require modification for every option that gets passed to every handler (only bearingSnap right now)

**Pull request merged**